src/main/java/org/eclipse/yasson/internal/serializer/MapDeserializer.java

```diff
@@ -89,7 +98,17 @@ public void appendResult(Object result) {
 
     @SuppressWarnings("unchecked")
     private <V> void appendCaptor(String key, V value) {
-        ((Map<String, V>) getInstance(null)).put(key, value);
+        if (Enum.class.isAssignableFrom((Class<?>) mapKeyRuntimeType)) {
+            Enum<?> enumKey = null;
+            try {
+                enumKey = (Enum<?>) ((Class<Enum<?>>) mapKeyRuntimeType).getMethod("valueOf", String.class).invoke(null, key);
```

Comment:
Using reflection here seemed hacky, but it's the best I could come up with as `Enum.valueOf(Class<T>, String value)` was not usable. Open to other suggestions on this part

Reply:
This worked for me:
```suggestion
                enumKey = Enum.valueOf((Class<Enum>) mapKeyRuntimeType, key);
```
It leaves `Class<Enum>` as a semi-raw type but could not get it to work otherwise

Reply:
thanks, I tried this several times and it always refused to compile, but in the set of changes I have in this PR it does indeed work now -- much cleaner